connmgr/connmanager.go

```diff
@@ -54,6 +53,9 @@ var (
 
 	// ErrPeerNotFound is an error that is thrown if the peer was not found.
 	ErrPeerNotFound = errors.New("peer not found")
+
+	//ErrAdressManagerNil is used to indicate that Address Manager cannot be nil in the configuration.
+	ErrAdressManagerNil = errors.New("Config: Address manager cannot be nil")
```

Comment:
It seems to me that all these errors are going to create a wrong stack trace. No?

Should somehow create these errors at the time they happened. Maybe just wrapping is enough. But must make sure